### PR TITLE
Add faunchpad and my keymap

### PR DESCRIPTION
**Pull request by loadedsith**
The Faunch pad had a limited release, as far as I can tell, it is very uncommon. Shortly after the release, something happened that caused the original author to not contribute the files to QMK. I do not know if my contributing them now is the right thing to do or not, but I fear the files could be lost if they're not contributed. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/gboards/k/faunchpad/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ *  '---------------------------'
+ */
+
+// Don't edit your keymap here! Do it in your dicts/ !
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT_faunch( 	KC_1,  KC_2,  KC_3,  KC_4,
+  										  KC_5,  KC_6,  KC_7,  KC_8)
+};
+
+// Don't fuck with this, thanks.
```
Comment (dunk2k) [inline keyboards/gboards/k/faunchpad/keymaps/default/keymap.c:38]:
```suggestion
// please do not alter this section of code, thanks.
```

File: keyboards/gboards/k/faunchpad/keymaps/grahampheath-kvm/keymap.c [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        } else {
+          // when keycode QMKBEST is released
+        }
+        break;
+
+      }
+    return true;
+};
+
+// Don't fuck with this, thanks.
```
Comment (dunk2k) [inline keyboards/gboards/k/faunchpad/keymaps/grahampheath-kvm/keymap.c:135]:
```suggestion
// please do not alter this section of code, thanks.
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/gboards/k/faunchpad/readme.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# FaunchPad!
+
+![Faunchpad](https://assets.bigcartel.com/product_images/247416671/IMG_20191025_1322045-01.jpeg)
```
Comment (dunk2k) [inline keyboards/gboards/k/faunchpad/readme.md:3]: Could this image be re-hosted on imgur?

File: keyboards/gboards/k/faunchpad/config_engine.h
Comment (waffle87) [inline keyboards/gboards/k/faunchpad/config_engine.h:1]: This file is missing a license header.

File: keyboards/gboards/k/faunchpad/config_engine.h [hunk 4]
```diff
@@ -0,0 +1,28 @@
+//#include "g/engine.h"
+
+// Configuration options for the engine
+
```
Comment (waffle87) [inline keyboards/gboards/k/faunchpad/config_engine.h:4]:
```suggestion
#pragma once
```

File: keyboards/gboards/k/faunchpad/faunchpad.c [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "faunchpad.h"
```
[the remark this answers is not in the document]
Comment (waffle87) [reply keyboards/gboards/k/faunchpad/faunchpad.c:16]: Edited.

File: keyboards/gboards/k/faunchpad/faunchpad.c [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+// These are only required if you want to perform custom actions.
+
+void matrix_scan_kb(void) {
+#ifdef DEBUG_MATRIX
+    for (uint8_t c = 0; c < MATRIX_COLS; c++)
+			for (uint8_t r = 0; r < MATRIX_ROWS; r++)
+		  	if (matrix_is_on(r, c)) xprintf("r:%d c:%d \n", r, c);
+#endif
+  matrix_scan_user();
+}
```
Comment (waffle87) [inline keyboards/gboards/k/faunchpad/faunchpad.c:29]: Does this serve any purpose beside printing debug info? If not, it's best if this file is just removed.

File: keyboards/gboards/k/faunchpad/info.json [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+        "vid": "0x1337",
+        "pid": "0x0006",
+        "device_version": "0.0.1"
+    },
+    "processor": "atmega32u4",
+    "bootloader": "atmel-dfu",
+    "width": 4,
+    "height": 2,
+    "layouts": {
+      "LAYOUT_faunch": {
```
Comment (waffle87) [inline keyboards/gboards/k/faunchpad/info.json:16]: This change will need to be reflected in the keymaps as well.
```suggestion
      "LAYOUT": {
```

File: keyboards/gboards/k/faunchpad/keymaps/grahampheath-kvm/keymap.c [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+    switch (keycode) {
+      case PORTONE:
+          if (record->event.pressed) {
+              // when keycode QMKBEST is pressed
+              // SEND_STRING(SS_TAP() SS_DELAY(600) SS_TAP(X_SCROLLLOCK) SS_DELAY(600) SS_TAP());
+              tap_code(KC_SCROLL_LOCK);
+              tap_code(KC_SCROLL_LOCK);
+              tap_code(KC_1);
+          } else {
+              // when keycode QMKBEST is released
```
Comment (waffle87) [inline keyboards/gboards/k/faunchpad/keymaps/grahampheath-kvm/keymap.c:61]:
```suggestion
              tap_code(KC_SCROLL_LOCK);
              tap_code(KC_SCROLL_LOCK);
              tap_code(KC_1);
```

File: keyboards/gboards/k/faunchpad/keymaps/grahampheath-kvm/keymap.c [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+      case PORTTHREE:
+        if (record->event.pressed) {
+          // when keycode QMKBEST is pressed
+          // SEND_STRING(SS_TAP(X_SCROLLLOCK) SS_DELAY(600) SS_TAP(X_SCROLLLOCK) SS_DELAY(600) SS_TAP(X_3));
+          tap_code(KC_SCROLL_LOCK);
+          tap_code(KC_SCROLL_LOCK);
+          tap_code(KC_3);
+          
+        } else {
+          // when keycode QMKBEST is released
```
Comment (waffle87) [inline keyboards/gboards/k/faunchpad/keymaps/grahampheath-kvm/keymap.c:87]:
```suggestion
          tap_code(KC_SCROLL_LOCK);
          tap_code(KC_SCROLL_LOCK);
          tap_code(KC_3);
```

File: keyboards/gboards/k/faunchpad/keymaps/grahampheath-kvm/keymap.c [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+      case NEXT:
+        if (record->event.pressed) {
+          // when keycode QMKBEST is pressed
+          // SEND_STRING(SS_TAP(X_SCROLLLOCK) SS_DELAY(600) SS_TAP(X_SCROLLLOCK) SS_DELAY(600) SS_TAP(X_PGDOWN));
+          tap_code(KC_SCROLL_LOCK);
+          tap_code(KC_SCROLL_LOCK);
+          tap_code(KC_PGDN);
+          
+        } else {
+          // when keycode QMKBEST is released
```
Comment (waffle87) [inline keyboards/gboards/k/faunchpad/keymaps/grahampheath-kvm/keymap.c:114]:
```suggestion
          tap_code(KC_SCROLL_LOCK);
          tap_code(KC_SCROLL_LOCK);
          tap_code(KC_PGDN);
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**